### DISCUSSION TITLE
Is adding <CR> at the end of the line for normal mode mappings required for all mappings?

### DISCUSSION THREAD

**Discussion by BillyGallagher** [esm7/obsidian-vimrc-support#243] [Q&A] [answered]
I just installed this plugin today and noticed the banner warning about needing to add \<CR\> at the end of the line for normal mode mappings. I've only just started setting this up, but from my testing it seems like it's only necessary to add \<CR\> after a command.

i.e. `nmap j gj` does not require \<CR\> at the end, but this example requires the \<CR\> on the second line:
```
exmap blankBelow obcommand obsidian-editor-shortcuts:insertLineBelow
nmap &b& :blankBelow<CR>
nmap o &b&i
```

So, is it just when mapping commands that need to be followed by \<CR\>? If so, is it all commands or is it specifically CodeMirror commands? (assuming it's related to the bug discussed in the readme).

**Comment by esm7** [comment, marked as answer]
Similarly to "real" Vim, only Ex commands (everything that starts with `:`) need the `<CR>`. I indeed see that the warning banner was not clear enough, thank you for pointing it out!

**Reply by BillyGallagher** [reply]
Thank you! Loving this plugin so far
